Remove libsbml refs

skeating opened the pull request:
## Description
<!--- Describe your changes in detail -->
changes many remaining libsbml references
and got it building on Windows

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here, using the 'fixes #<issue>' syntax. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [ ] I have updated all documentation necessary.
- [ ] I have checked spelling in (new) comments.

## Testing
- [ ] Testing is done automatically and codecov shows test coverage
- [ ] This cannot be tested automatically <!-- describe how it has been tested -->

skeating committed:
remove reference to libSBML

skeating committed:
replacements of libSBML

skeating committed:
get all files included in IDE

skeating committed:
change libsbml to liblx

skeating committed:
remove repeat extern declaration

skeating commented:
Hi Matthew

If you could check that this still builds for you.

I solved by dllimport issue and I think I told cmake about bindings so it may help with that too.

mattagape commented on CMakeLists.txt:
```diff
@@ -1,11 +1,11 @@
 ###############################################################################
 #
-# Description       : CMake build script for libSBML
+# Description       : CMake build script for libLX
```
I believe the following lines in this CMakeLists.txt need to be updated: 139, 1482, 1495, 1505 (I'm not sure about the last two)

mattagape commented on src/CMakeLists.txt:
```diff
@@ -102,22 +91,76 @@ configure_file(
+endforeach(dir)
+
+
+file(GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/sbml/*.h")
```
Mentions sbml.

mattagape commented on src/CMakeLists.txt:
```diff
@@ -102,22 +91,76 @@ configure_file(
+
+
+file(GLOB files "${CMAKE_CURRENT_SOURCE_DIR}/sbml/*.h")
+install(FILES ${files} DESTINATION include/sbml)
```
also mentions sbml

mattagape commented on src/CMakeLists.txt:
```diff
@@ -365,46 +408,67 @@ endif()
+#
+
+file(GLOB LIBLX_HEADER_FILES "${CMAKE_CURRENT_BINARY_DIR}/liblx/common/*.h")    
+foreach(file ${LIBSBML_SOURCES})
```
Please check if we still need this. Also, line 231, do we need it?

mattagape commented on src/bindings/swig/OStream.h:
```diff
@@ -90,7 +90,7 @@
 #include <sstream>
 
-#include <liblx/xml/XMLExtern.h>
+#include <liblx/xml/common/extern.h>
```
Some or all of the following lines need changing, I believe:
7,8,34,42,43,60,62,77,79

mattagape commented on src/liblx/xml/LibXMLTranscode.h:
```diff
@@ -43,7 +43,7 @@
 #include <libxml/parser.h>
-#include <liblx/xml/XMLExtern.h>
+#include <liblx/xml/common/extern.h>
 
```
query lines 2,9,10,36

fbergmann committed:
- dont configure files that are not needed

(potentially rethink the use of the doxygen approach, in favor of something easier)

fbergmann committed:
- allow ctest to compile

mattagape reviewed (approved):
There are a few instances of "sbml" terms still floating around, but I think we can mop those up later.
